src/Iris/Tool.hs

```diff
@@ -132,6 +136,32 @@ checkTool cmd Tool{..} = findExecutable (Text.unpack toolName) >>= \case
                 toolVersionOutput <- readProcess exe [Text.unpack versionArg] ""
                 let version = Text.strip $ Text.pack toolVersionOutput
 
-                if toolSelectorFunction cmd version
+                if toolSelectorFunction version
                 then pure ToolOk
                 else pure $ ToolCheckError $ ToolWrongVersion version
+
+newtype ToolCheckException = ToolCheckException ToolCheckError
+    deriving stock
+        ( Show  -- ^ @since 0.0.0.0
+        )
+
+    deriving newtype
+        ( Eq  -- ^ @since 0.0.0.0
+        )
+
+    deriving anyclass
+        ( Exception  -- ^ @since 0.0.0.0
+        )
+
+
+{-|
+
+__Throws:__ 'ToolCheckException' if can't find a tool or if it has wrong version.
+@since 0.0.0.0
+-}
+need :: [Tool] -> IO ()
```

Comment:
I don't know if this is the best signature for the `need` function. Maybe we should add a `MonadIO m` constraint? Let me know, please :)

Reply:
Agree, let's actually have `MonadIO m =>` constraint here. This function is supposed to be called from inside the custom monad, so having the `MonadIO` constraint allows us to do this easily without the need to call `liftIO`.